backend/app/models/shift_assignments.py

```
@@ -9,11 +9,16 @@ class ShiftAssignments(Base):
+    effective_to = Column(Date, nullable=True, default=None)
     is_active = Column(Boolean, default=True)
     created_at = Column(DateTime(timezone=True), server_default=func.current_timestamp())
     updated_at = Column(DateTime(timezone=True), server_default=func.current_timestamp(), onupdate=func.current_timestamp())
+    deleted_at = Column(DateTime(timezone=True), nullable=True)
     
     __table_args__ = (
         CheckConstraint("effective_to IS NULL OR effective_to >= effective_from", name="assignment_dates_valid"),
+        CheckConstraint("deleted_at IS NULL OR is_active = FALSE", name="soft_delete_check"),
+        Index('idx_shift_assignments_user_id', 'user_id'),
+        Index('idx_shift_assignments_pattern_id', 'pattern_id'),
+        Index('idx_current_shift_assignments', 'user_id', 'pattern_id', postgresql_where=Column('is_active') == True),
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `Column('is_active'):` for truth checks

Replace with `Column('is_active')`

(E712)